obspy/core/stream.py

```diff
@@ -3118,6 +3119,85 @@ def remove_sensitivity(self, *args, **kwargs):
+        """
+        Return stream with traces stacked by the same selected metadata.
+
+        :param group: Stack waveforms together which have the same metadata
+            given by this parameter. The parameter should name the
+            corresponding keys of the stats object,
+            e.g. `'{network}.{station}'` for stacking all
+            locations and channels of the stations and returning a stream
+            consisting of one stacked trace for each station.
+            This parameter can take two special values,
+            `'seedid'` which stacks the waveforms by seedid and
+            `'all'` (default) which stacks together all traces in the stream.
+        :param type: Type of stack, one of the following:
+            `'linear'`: average stack (default),
+            `('pw', order)`: phase weighted stack of given order,
+            see [Schimmel1997]_,
+            `('root', order)`: root stack of given order.
+        :param npts_tol: Tolerate traces with different number of points
+            with a difference up to this value. Surplus samples are discarded.
+
+        :returns: New stream object with stacked traces. The metadata of each
+            trace corresponds to the metadata of the original traces
+            if those are the same. Additionaly, the entries
+            `stack` (result of the format operation on the group parameter) and
+            `stack_count` (number of stacked traces)
+            are written to the stats object.
+
+        >>> from obspy import read
+        >>> st = read()
+        >>> stack = st.stack()
+        >>> print(stack)  # doctest: +ELLIPSIS
+        1 Trace(s) in Stream:
+        BW.RJOB.. | 2009-08-24T00:20:03.000000Z - ... | 100.0 Hz, 3000 samples
+        """
+        if group == 'seedid':
+            group = '{network}.{station}.{location}.{channel}'
+        if len(type) == 2:
+            type, order = type
+        groups = collections.defaultdict(list)
+        for tr in self:
+            groups[group.format(**tr.stats)].append(tr)
+        stacks = []
+        for groupid, grouptrcs in groups.items():
+            header = {k: v for k, v in grouptrcs[0].stats.items()
+                      if all(tr.stats.get(k) == v for tr in grouptrcs)}
+            header['stack'] = groupid
+            header['stack_count'] = len(grouptrcs)
+            npts_all = [len(tr) for tr in grouptrcs]
+            npts_dif = max(npts_all) - min(npts_all)
+            npts = min(npts_all)
+            if npts_dif > npts_tol:
+                msg = ('Difference of number of points of the traces is higher'
+                       ' than requested tolerance ({} > {})')
+                raise ValueError(msg.format(npts_dif, npts_tol))
+            elif npts_dif > 0:
+                grouptrcs = [copy.copy(tr) for tr in grouptrcs]
+                for tr in grouptrcs:
+                    tr.data = tr.data[:npts]
+            data = np.array([tr.data for tr in grouptrcs])
```

Comment:
You could
```suggestion
            data = np.array([tr.data[:npts] for tr in grouptrcs])
```
and not bother with copying above.